werkzeug/serving.py

```diff
@@ -144,6 +144,13 @@ def shutdown_server():
         if request_url.scheme and request_url.netloc:
             environ['HTTP_HOST'] = request_url.netloc
 
+        try:
+            peer_cert = self.connection.getpeercert()
+            if peer_cert is not None:
+                environ['SSL_CLIENT_CERT'] = peer_cert
+        except ValueError:
```

Comment:
What is this intended to catch? It notably doesn't catch the `AttributeError` if `self.connection` is not an SSL socket, which breaks plain HTTP serving.

Reply:
According to the [`ssl` lib's documentation](https://docs.python.org/3/library/ssl.html#ssl.SSLSocket.getpeercert), `SSLSocket.getpeercert` can only raise a _ValueError_: whenever the SSL handshake is not yet done. This is what is being catched here. 
About the `AttributeError`, what do you think it is the best way to treat an exception in this case?

Reply:
`ValueError`: I don't think it is possible to get here if we haven't finished the SSL handshake, since we've already transferred data.

`AttributeError` is expected if this is a non-SSL request, so I think should be silently ignored.